qemu/tests/migration_with_numa.py

```diff
@@ -0,0 +1,113 @@
+            params['host-nodes_mem_%s' % mem_dev] = str(host_nodes_size[0][0])
+            del host_nodes_size[0]
+        else:
+            test.cancel("host nodes do not have enough memory for testing!")
+
+    params['start_vm'] = 'yes'
+    env_process.preprocess(test, params, env)
+    vm = env.get_vm(params["main_vm"])
+    session = vm.wait_for_login()
+    # do migration
+    mig_timeout = float(params.get("mig_timeout", "3600"))
+    mig_protocol = params.get("migration_protocol", "tcp")
+    vm.migrate(mig_timeout, mig_protocol, env=env)
+    session = vm.wait_for_login()
+
+    os_type = params["os_type"]
+    if os_type == 'linux':
+        error_context.context("Check the memory in guest", logging.info)
+        # Use 30 plus the gap of 'MemTotal' in OS and '-m' in cli as threshold
+        mem_total = get_mem_info(session, 'MemTotal')
+        mem_total = float(normalize_data_size('%s KB' % mem_total))
+        error_context.context("MemTotal in guest os is %s MB"
+                              % mem_total, logging.info)
+        threshold = float(params.get_numeric("mem") - mem_total) + 30
+        error_context.context("The acceptable threshold is: %s"
+                              % threshold, logging.info)
+        guest_nodes_size = get_nodes_size(size_type='MemTotal', session=session)
+        guest_nodes_size = dict(guest_nodes_size)
+        for nodenr, node in enumerate(params.objects('guest_numa_nodes')):
+            mdev = params.get("numa_memdev_node%d" % nodenr)
+            if mdev:
+                mdev = mdev.split('-')[1]
+                size = float(normalize_data_size(params.get("size_mem_%s" % mdev)))
+                if size - guest_nodes_size[nodenr] > threshold:
+                    test.fail("[Guest]Wrong size of numa node %d: %f. Expected:"
+                              " %s" % (nodenr, guest_nodes_size[nodenr], size))
+
+    error_context.context("Check the memory in dest host", logging.info)
+    qemu_pid = vm.get_pid()
+    for mem_dev in mem_devs:
+        memdev_params = params.object_params(mem_dev)
+        size_mem = memdev_params.get('size_mem')
+        size_mem = int(float(normalize_data_size(size_mem, 'K')))
+        smaps = process.getoutput("grep -E -B1 '^Size: *%d' /proc/%d/smaps"
+                                  % (size_mem, qemu_pid))
+        mem_start_pattern = r'(\w+)-\w+\s+\w+-\w+\s+\w+\s+\w+:\w+\s\w+\s+\n'\
+                            r'Size:\s+%d' % size_mem
+        match = re.search(mem_start_pattern, smaps)
+        if not match:
+            test.error("Failed to get the mem start in smaps: %s" % smaps)
+        mem_start = match.groups()[0]
+        numa_maps = process.getoutput("grep %s /proc/%d/numa_maps"
+                                      % (mem_start, qemu_pid))
+        node_match = re.search(r'bind:(\d+)', numa_maps)
+        if not node_match:
+            test.error("Failed to get the bind node in numa_maps: %s" % numa_maps)
```

Comment:
test.fail? If bind can't be found, then the policy may not take effect.

Reply:
Make sense, updated, thanks!